crates/dht/src/time.rs

```diff
@@ -0,0 +1,1123 @@
+//! Partition of time into slices.
+//!
+//! Provides an encapsulation for calculating time slices that is consistent when computed
+//! by different nodes. This is used to group Kitsune2 op data into time slices for combined hashes.
+//! Giving multiple nodes the same time slice boundaries allows them to compute the same combined
+//! hashes, and therefore effectively communicate which of their time slices match and which do not.
+//!
+//! A time slice is defined to be an interval of time that is closed at the start and
+//! open at the end. That is, an interval [start, end)` where:
```

Comment:
```suggestion
//! open at the end. That is, an interval [start, end] where:
```

not sure if its a typo `or` notation

Reply:
It's intentional [notation](https://www.statisticshowto.com/calculus-definitions/closed-interval-open-interval/) but I did mean to match the backticks, fixed